bpf/gotracer/go_grpc.c

```diff
@@ -592,32 +610,76 @@ int beyla_uprobe_transport_http2Client_NewStream(struct pt_regs *ctx) {
         }
 
 #ifndef NO_HEADER_PROPAGATION
-        u32 next_id = 0;
-        // Read the next stream id from the httpClient
-        bpf_probe_read(
-            &next_id,
-            sizeof(next_id),
-            (void *)(t_ptr + go_offset_of(ot, (go_offset){.v = _http2_client_next_id_pos})));
-
-        bpf_dbg_printk("next_id %d", next_id);
+        bpf_dbg_printk("conn_ptr %llx", conn_ptr_key);
 
         grpc_client_func_invocation_t *invocation =
             bpf_map_lookup_elem(&ongoing_grpc_client_requests, &g_key);
 
-        if (invocation) {
-            grpc_client_func_invocation_t inv_save = *invocation;
-            // This map is an LRU map, we can't be sure that all created streams are going to be
-            // seen later by writeHeader to clean up this mapping.
-            bpf_map_update_elem(&ongoing_streams, &next_id, &inv_save, BPF_ANY);
+        if (invocation && conn_ptr_key) {
+            transport_new_client_invocation_t wrapper = {};
+            wrapper.inv = *invocation;
+            wrapper.s_key.stream_id = 0;
```

Comment:
We use a combination of the connection pointer and the stream_id now as a key, rather than just the stream_id.

Reply:
We used to read the stream_id here before from the nextID field on the client, but as per Bug report item 2) this is not correct.

Reply:
Instead we read the stream id after the stream is created with the return probe on NewStream